sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -125,12 +129,37 @@ case class InMemoryTableScanExec(
     val schema = relation.partitionStatistics.schema
     val schemaIndex = schema.zipWithIndex
     val relOutput: AttributeSeq = relation.output
-    val buffers = relation.cachedColumnBuffers
+    val partitionFilter = newPredicate(
+      partitionFilters.reduceOption(And).getOrElse(Literal(true)),
+      schema)
+
+    val buffers = if (inMemoryPartitionPruningEnabled && !relation.batchStats.value.isEmpty) {
+      val validPartitions = relation.batchStats.value.asScala
+        .filter(batchStat => partitionFilter(batchStat._2))
+        .map(_._1)
+        .distinct
+      if (validPartitions.isEmpty) {
+        new EmptyRDD[CachedBatch](sparkContext)
+      } else {
+        new PartitionPruningRDD[CachedBatch](relation.cachedColumnBuffers,
```

Comment:
Cool yeah used this (and removed the EmptyRDD constructor as well). I'd prefer to keep the logging in the function though.